core/matrix/dense.cpp

```diff
@@ -468,10 +468,9 @@ void Dense<ValueType>::move_to(Dense<next_precision<ValueType>> *result)
 template <typename ValueType>
 void Dense<ValueType>::convert_to(Coo<ValueType, int32> *result) const
 {
-    conversion_helper(
-        result, this,
-        dense::template make_convert_to_coo<const Dense<ValueType> *&,
-                                            decltype(result)>);
+    conversion_helper(result, this, [](const auto &in, const auto &out) {
```

Comment:
Note: those are `const` references to a `const` and a non-`const` pointer.

Reply:
Do you try the original one with current switch implementation?

Reply:
It still doesn't work, since it can't deduce the `auto` return type, and thus the type of the parameter correctly.